x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -2179,14 +2199,21 @@ export class RulesClient {
         this.updateMeta({
           ...attributes,
           enabled: false,
-          scheduledTaskId: null,
+          scheduledTaskId: attributes.scheduledTaskId === id ? attributes.scheduledTaskId : null,
           updatedBy: await this.getUserName(),
           updatedAt: new Date().toISOString(),
         }),
         { version }
       );
+
```

Comment:
This update to the disable logic takes into account that there may be pre 8.0 rules running where the scheduled task ID does not equal the rule ID. In these cases, we still want to remove the task document so that a new one matching the rule id can be created on enable.

If the scheduledTaskId already matches the rule ID, this will set the task to disabled.